### ISSUE TITLE
react/prop-types false negative for dynamic object access?

### ISSUE THREAD

**Issue by amilajack** [yannickcr/eslint-plugin-react#818]
``` js
class Example extends Component {
  examplFn() {
    const type = 'items';
    const { options } = this.props[type] // <-- items.options is missing in props validation
  }
  // ...
}


Example.propTypes = {
  items: PropTypes.shape({
    options: PropTypes.arrayOf(PropTypes.string)
  }).isRequired
}
```

I'm assuming this isn't by design.

**Comment by lencioni** [comment]
I think we could probably fix this for simple cases, but more complex cases will either be impossible or very difficult. I'd be happy to review a PR that fixes this, but I'm not entirely sure it is worth the effort.

**Comment by ljharb** [comment]
PRs that improve propTypes detection are exceedingly welcome, but I don't think leaving this issue open is going to help the likelihood of that.
